lib/core/test/unit/Cardano/Wallet/Primitive/CoinSelection/MA/RoundRobinSpec.hs

```diff
@@ -567,17 +602,68 @@ genSelectionCriteria genUTxOIndex = do
           )
         ]
     extraCoinSource <- oneof [ pure Nothing, Just <$> genCoinSmall ]
+    assetsToBurn <- TokenMap.fromFlatList <$>
```

Comment:
As discussed with @jonathanknowles , I do not think these generators sufficiently cover the potential "universe" of SelectionCriteria. They unnecessarily make the set of user-specified assets and the set of assets to burn disjoint.

I made a hacky PR to see how the tests would behave if I altered the generator: https://github.com/input-output-hk/cardano-wallet/pull/2740

Reply:
I've added a set of coverage checks to cover several possibilities in f7a93280fa4b3cbe0d070846351720fa02fc4435. I think this covers all the possibilities we're interested in.

Out of these coverage checks, only one is not satisfied (namely, that we occasionally generate minted values that are not spent or burned). I've fixed this in f8d177742fabe84bfbe4966886581042dbd87f2f.

However, this revealed a faulty assertion in `onOutputsInsufficient`, which I have fixed in 36b2be9432f87e89a26995e3985677120e7cf266.